secant/Util/RecoveryPhraseStorage.swift

```diff
@@ -0,0 +1,229 @@
+//
+//  RecoveryPhraseStorage.swift
+//  secant-testnet
+//
+//  Created by Lukáš Korba on 03/10/2022.
+//
+
+import Foundation
+
+/// Zcash implementation of the keychain that is not universal but designed to deliver functionality needed by the wallet itself.
+/// All the APIs should be thread safe according to official doc:
+/// https://developer.apple.com/documentation/security/certificate_key_and_trust_services/working_with_concurrency?language=objc
+/// For seamless usage the interface is designed to deal with simple types:
+final class RecoveryPhraseStorage {
+    enum KeychainError: Error, Equatable {
+        case duplicate
+        case unknown(OSStatus)
+        case encoding
+        case decoding
+        case noDataFound
+    }
+
+    static let `default`:RecoveryPhraseStorage = RecoveryPhraseStorage()
+
+    enum Constants {
+        static let zcashRecoveryPhrase = "zcashRecoveryPhrase"
+        static let zcashBirthday = "zcashBirthday"
+    }
+    
+    // MARK: - Recovery Phrase Interface
+    
+    private func setRecoveryPhrase(_ phrase: String) throws {
```

Comment:
Would a private extension help clean this file up some, instead of private functions?

Reply:
done